src/cdk-lib/bedrock/agent.ts

```diff
@@ -596,6 +614,130 @@ export class Agent extends Construct {
   }
 }
 
+/**
+ * Deploy a Bedrock Agent.
+ */
+export class Agent extends AgentBase {
+  public readonly name: string;
+  public readonly role: iam.Role;
+  public readonly agentId: string;
+  public readonly agentInstance: bedrock.CfnAgent;
+  public readonly agentArn: string;
+  public readonly aliasId?: string;
+  public readonly aliasArn?: string;
+  public readonly aliasName?: string;
+  public readonly agentVersion: string;
+  public actionGroups: bedrock.CfnAgent.AgentActionGroupProperty[] = [];
+  public knowledgeBases: IKnowledgeBase[] = [];
+
+  constructor(scope: Construct, id: string, props: AgentProps) {
+    super(scope, id);
+    validatePromptOverrideConfiguration(props.promptOverrideConfiguration);
+
+    validateModel(props.foundationModel);
+
+    this.name = props.name ?? generatePhysicalNameV2(
+      this,
+      'bedrock-agent',
+      { maxLength: 32, lower: true, separator: '-' });
+
+    if (props.existingRole) {
+      this.role = props.existingRole;
+    } else {
+      this.role = new iam.Role(this, 'Role', {
+        assumedBy: new iam.ServicePrincipal('bedrock.amazonaws.com'),
+        roleName: generatePhysicalNameV2(
```

Comment:
we need to use the new capability of generatePhysicalNameV2 here to avoid collisions with name generation